addons/osio-addon/src/main/java/io/fabric8/launcher/osio/preparers/InsertMavenDependenciesPreparer.java

```diff
@@ -0,0 +1,70 @@
+package io.fabric8.launcher.osio.preparers;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import io.fabric8.launcher.base.maven.Maven;
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBooster;
+import io.fabric8.launcher.core.api.ProjectileContext;
+import io.fabric8.launcher.core.spi.ProjectilePreparer;
+import io.fabric8.launcher.osio.projectiles.context.OsioProjectileContext;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Dependency;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class InsertMavenDependenciesPreparer implements ProjectilePreparer {
+
+    private static final Logger log = Logger.getLogger(InsertMavenDependenciesPreparer.class.getName());
+
+
+    @Override
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof OsioProjectileContext)) {
+            return;
+        }
+        OsioProjectileContext createProjectileContext = (OsioProjectileContext) context;
+        Path pom = projectPath.resolve("pom.xml");
+        List<String> dependencies = createProjectileContext.getDependencies();
+
+        // Perform model changes
+        if (Files.isRegularFile(pom) && !dependencies.isEmpty()) {
+            try {
+                Model model = Maven.readModel(pom);
+                List<String> boosterDependencies = model.getDependencies().stream().map(d ->
+                        (d.getGroupId() + ":" + d.getArtifactId())
+                ).collect(Collectors.toList());
+                // Get the list of dependencies in the format of 'groupId:artifactId:version'
+                for (String dep : dependencies) {
+                    if (!isDuplicateDependency(boosterDependencies, dep)) {
+                        Dependency dependency = new Dependency();
+                        String[] groupIdArtifactIdVersion = dep.split(":");
+                        dependency.setGroupId(groupIdArtifactIdVersion[0]);
+                        dependency.setArtifactId(groupIdArtifactIdVersion[1]);
+                        dependency.setVersion(groupIdArtifactIdVersion[2]);
+                        model.addDependency(dependency);
+                    }
+                }
+                Maven.writeModel(model);
+            } catch (Exception e) {
+                log.log(Level.SEVERE, "An exception occurred while adding the dependencies. " + e.toString());
+            }
+
+        }
+    }
+
+    public boolean isDuplicateDependency(List<String> allDependencies, String dependency) {
+        try {
+            String[] groupIdArtifactIdVersion = dependency.split(":");
+            return allDependencies.contains(groupIdArtifactIdVersion[0] + ":" + groupIdArtifactIdVersion[1]);
```

Comment:
This will return false if the version is sent in the list of dependencies

Reply:
yes, It will return false but here I'm sending list of dependencies in the format `[ "groupId:artifactId", "groupId:artifactid"]`